sdlib/d/gc/tcache.d

```diff
@@ -54,6 +64,77 @@ public:
 		pd.arena.free(emap, pd, ptr);
 	}
 
+	/**
+	 * Appendables API.
+	 */
+
+	// Get the capacity of the array slice denoted by slice and length.
+	// It includes the space occupied by the slice itself.
+	size_t getArrayCapacity(void* sliceAddr, size_t sliceLen) {
+		if ((sliceAddr is null) || (!sliceLen))
```

Comment:
Why would a slice of length 0 would necessarily have 0 capacity? Also, I don't see why you need to special case null, as the page descriptor will be empty.